packages/zero-client/src/client/zero.ts

```diff
@@ -477,7 +478,9 @@ export class Zero<
             mutatorOrMutators,
             schema,
             slowMaterializeThreshold,
-          );
+            // Replicache expects mutators to only be able to return JSON
+            // but Zero wraps the return with: `{server?: Promise<MutationResult>, client?: T}`
+          ) as () => MutatorReturn;
```

Comment:
This is unfortunate. We could relax the Replicache return type constraint on mutators.